front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -228,193 +199,193 @@ export const usePathfinding = (
-
-        if (hasInvalidPathItems) {
-          handleInvalidPathItems(pathSteps, pathfindingResult.items);
-          return;
-        }
+      if (hasInvalidPathItems) {
+        handleInvalidPathItems(pathSteps, pathfindingResult.items);
+        return;
+      }
 
-        let error: string;
-        if (pathfindingResult.failed_status === 'internal_error') {
-          const translationKey = pathfindingResult.core_error.type.startsWith('core:')
-            ? pathfindingResult.core_error.type.replace('core:', '')
-            : pathfindingResult.core_error.type;
-          error = t(`coreErrors.${translationKey}`, {
-            defaultValue: pathfindingResult.core_error.message,
-          });
-        } else {
-          error = t(`pathfindingErrors.${pathfindingResult.error_type}`);
-        }
-        pathfindingDispatch({
-          type: 'PATHFINDING_ERROR',
-          message: error,
+      let error: string;
+      if (pathfindingResult.failed_status === 'internal_error') {
+        const translationKey = pathfindingResult.core_error.type.startsWith('core:')
+          ? pathfindingResult.core_error.type.replace('core:', '')
+          : pathfindingResult.core_error.type;
+        error = t(`coreErrors.${translationKey}`, {
+          defaultValue: pathfindingResult.core_error.message,
         });
-      } catch (e) {
-        if (isObject(e)) {
-          let error;
-          if ('error' in e) {
-            dispatch(setFailure(castErrorToFailure(e, { name: t('pathfinding') })));
-            error = 'failedRequest';
-          } else if ('data' in e && isObject(e.data) && 'message' in e.data) {
-            error = e.data.message as string;
-            if (e.data.message === 'Infra not loaded' || e.data.message === 'Invalid version') {
-              setIsInfraError(true);
-            }
+      } else {
+        error = t(`pathfindingErrors.${pathfindingResult.error_type}`);
+      }
+      pathfindingDispatch({
+        type: 'PATHFINDING_ERROR',
+        message: error,
+      });
+    } catch (e) {
+      if (isObject(e)) {
+        let error;
+        if ('error' in e) {
+          dispatch(setFailure(castErrorToFailure(e, { name: t('pathfinding') })));
+          error = 'failedRequest';
+        } else if ('data' in e && isObject(e.data) && 'message' in e.data) {
+          error = e.data.message as string;
+          if (e.data.message === 'Infra not loaded' || e.data.message === 'Invalid version') {
+            setIsInfraError(true);
           }
-          pathfindingDispatch({ type: 'PATHFINDING_ERROR', message: error });
         }
+        pathfindingDispatch({ type: 'PATHFINDING_ERROR', message: error });
       }
-    };
-
-    if (infra && infra.state === 'CACHED' && pathfindingState.mustBeLaunched) {
-      startPathFinding();
     }
-  }, [pathfindingState.mustBeLaunched, infra]);
+  };
 
-  useEffect(() => setIsPathfindingInitialized(true), []);
+  useEffect(() => {
+    launchPathfinding();
```

Comment:
Previously we'd re-launch the pathfinding when the infra state switches from loading to cached. Now we don't do this anymore. Do we want to retain this behavior?

Reply:
Yes you're right, we should keep this behavior